server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -2312,32 +2312,29 @@ public SeedSystemVMTemplateResponse seedSystemVMTemplate(HashSet<String> imageSt
 
             startInstallTemplate(template.getId(), cmd.getId());
             // Decompress file
-            decompressFile(inputFile, "/tmp/" + template.getUuid() + "." + fileExtension);
+            boolean decompressed = decompressFile(inputFile, "/tmp/" + template.getUuid() + "." + fileExtension);
+            if (decompressed) {
+                inputFile = "/tmp/" + template.getUuid() + "." + fileExtension;
+            }
 
-            String finalDestination = mountPoint + "/template/tmpl/1/" + template.getId() + "/"+ template.getUuid() + "." + fileExtension;
+            String finalDestination = mountPoint + "/template/tmpl/2/" + template.getId() + "/"+ template.getUuid() + "." + fileExtension;
```

Comment:
I would use a getUserId() kind of mech here, as admin may not be the only on with ROOTadmin rights.